coverage.cmake

```diff
@@ -0,0 +1,45 @@
+#
+#   Copyright 2022 CNRS
+#
+#   Author: Guilhem Saurel
+#
+#
+
+#.rst:
+# .. ifmode:: internal
+#
+#   .. variable:: ENABLE_COVERAGE
+#
+#      When this is ON, coverage compiler flags are enabled
+
+option(ENABLE_COVERAGE "enable C++ and python code coverage" OFF)
+
+macro(_SETUP_COVERAGE)
+  if(ENABLE_COVERAGE)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
+    message(STATUS "Appending code coverage compiler flags: --coverage")
+    #add_custom_target(py_coverage
+      #COMMAND ${PYTHON_EXECUTABLE} -m coverage report)
+  endif()
+endmacro()
+
+macro(_SETUP_COVERAGE_FINALIZE)
+  if(ENABLE_COVERAGE)
+    set(_COVERAGE_DIR "${CMAKE_BINARY_DIR}/coverage")
+
+    # use lcov to gether c++ coverage reports from sources,
+    # and coverage.py to gather python coverage reports from installed files,
+    # and then generate HTML by removing the source prefix and python sitelib install prefix
+    add_custom_target(coverage
+      COMMAND lcov --include "${PROJECT_SOURCE_DIR}/\\*" -c -d ${PROJECT_SOURCE_DIR} -o cpp.lcov
+      COMMAND ${PYTHON_EXECUTABLE} -m coverage combine
+      COMMAND ${PYTHON_EXECUTABLE} -m coverage lcov -o python.lcov
```

Comment:
Maybe we don't want/need to run this if there's no Python code or Python tests or the first `lcov` command if there's no C++ coverage, maybe (in `_SETUP_COVERAGE`):
```
set_property(GLOBAL PROPERTY JRL_CMAKEMODULES_HAS_CPP_COVERAGE OFF)
set_property(GLOBAL PROPERTY JRL_CMAKEMODULES_HAS_PYTHON_COVERAGE OFF)
```

Then these properties are set to `ON` if coverage is added to a target (C++ side) or if Python tests are added

Reply:
fixed in #509, but by checking if `PYTHON_EXECUTABLE` is defined. But your solution looks actually better. I'll implement that.